python/lsst/obs/pfs/pfsMapper.py

```diff
@@ -173,6 +176,128 @@ def _shiftAmpPixels(self, rawExp):
 
             imArr[yslice, xslice] = ampPixels.reshape(ampshape)
 
+    @staticmethod
+    def lookupHdu(read, imType="IMAGE", md=None):
+        """Return the appropriate HDU
+
+        Parameters
+        ----------
+        read: `int`
+            The desired read of the device (1-indexed)
+        imType: `str`
+            IMAGE or REF
+        md: `lsst.daf.base.PropertyList`
+            Header from file (optional)
+            Used to validate read, and handle extra reset HDUs if present
+        """
+
+        if md:
+            nread = md.get("W_H4NRED")
+
+            if nread is not None and not (1 <= read <= nread):
+                raise RuntimeError(f"The read must be in the range 1..{nread}; saw {read}")
+
+        hdu = 2*read - 1 + dict(IMAGE=0, REF=1)[imType]
+
+        return hdu
+
+    def bypass_datasetExists_raw(self, datasetType, pythonType, butlerLocation, dataId):
+        return self.bypass_raw(datasetType, pythonType, butlerLocation, dataId, checkExistence=True)
+
+    def bypass_raw(self, datasetType, pythonType, butlerLocation, dataId, checkExistence=False):
+        if dataId['arm'] in "bmr":
+            raise IOError("Please ignore this bypass function for [bmr] data")
+
+        def readData(datasetType, butlerLocation, dataId, hdu=1, getMetadata=False, checkExistence=False):
```

Comment:
Docstring.

Reply:
It's a local function.  I added a note to this effect, but I don't think that a full docstring is appropriate.